### ISSUE TITLE
MPS: transformers crashing in executeMPSGraph with Function square_i64 was not found in the library

### ISSUE THREAD

**Issue by przem8k** [pytorch/pytorch#114532]
### 🐛 Describe the bug

I'm hitting a crash in executeMPSGraph when trying to run this [notebook](https://www.kaggle.com/code/jhoward/getting-started-with-nlp-for-absolute-beginners/notebook) on my M1 Macbook (smaller self-contained repro below) crashing in trainer.train() with the following error:

```
2023-11-25 09:29:20.582 Python[15924:310024] Error getting visible function:
 (null) Function square_i64 was not found in the library
/AppleInternal/Library/BuildRoots/495c257e-668e-11ee-93ce-926038f30c31/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Utility/MPSKernelDAG.mm:805: failed assertion `Error getting visible function:
```

To repro:

```python
import pandas as pd
from datasets import Dataset
from transformers import AutoModelForSequenceClassification, AutoTokenizer
from io import StringIO
from transformers import TrainingArguments, Trainer

sample_data = """input,labels
bazinga,0
please-just-work,1
"""

df = pd.read_csv(StringIO(sample_data))
ds = Dataset.from_pandas(df)

model_name = "microsoft/deberta-v3-small"
tokenizer = AutoTokenizer.from_pretrained(model_name)

def tokenizer_func(x): return tokenizer(x["input"])
ds_tokenized = ds.map(tokenizer_func, batched=True)
dds = ds_tokenized.train_test_split(0.2, seed=42)

bs = 16
epochs = 4
lr = 8e-5

args = TrainingArguments(
    "outputs",
    learning_rate=lr,
    warmup_ratio=0.1,
    lr_scheduler_type="cosine",
    fp16=False,
    evaluation_strategy="epoch",
    per_device_train_batch_size=bs,
    per_device_eval_batch_size=bs * 2,
    num_train_epochs=epochs,
    weight_decay=0.01,
    report_to="none",
)

model = AutoModelForSequenceClassification.from_pretrained(model_name, num_labels=1)
trainer = Trainer(
    model,
    args,
    train_dataset=dds["train"],
    eval_dataset=dds["test"],
    tokenizer=tokenizer,
)

# This crashes
trainer.train()
```

Looks similar to #78019 but the missing function is different.

Stack trace:

```
Thread 0 Crashed::  Dispatch queue: metal gpu stream
0   libsystem_kernel.dylib        	       0x18d17911c __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x18d1b0cc0 pthread_kill + 288
2   libsystem_c.dylib             	       0x18d0c0a40 abort + 180
3   libsystem_c.dylib             	       0x18d0bfd30 __assert_rtn + 284
4   Metal                         	       0x19744c3fc MTLReportFailure.cold.1 + 48
5   Metal                         	       0x197429380 MTLReportFailure + 464
6   MPSCore                       	       0x1981bed38 0x19818f000 + 195896
7   MPSCore                       	       0x19819569c 0x19818f000 + 26268
8   MPSCore                       	       0x198191c20 MPSKernelDAG::getDAGAndHash(id<MTLLibrary>, MPSDAGKernelOp*, NSMutableDictionary<NSString*, id<MTLFunction>>*, NSMutableArray<NSString>*, unsigned long long vector[4]&, bool*) + 456
9   MPSCore                       	       0x198193e7c 0x19818f000 + 20092
10  MPSCore                       	       0x1981eb8e4 MPSLibrary::CreateUberShaderKey(NSString*, MPSFunctionConstantList const&, unsigned long long, id<MTLFunction> (*)(id<MTLLibrary>, MPSKernelInfo const*, MPSFunctionConstantList const&, MPSFunctionConstructorExtraParams const&, NSError**), unsigned long long, MPSThreadGroupSizeMultipleType, MPSDriverCompilerOptions, MPSKernelDAGObject*, NSString*, unsigned long) + 660
11  MPSNDArray                    	       0x199452d44 0x1993fa000 + 363844
12  MPSNDArray                    	       0x19943f814 0x1993fa000 + 284692
13  MPSNDArray                    	       0x199440df8 0x1993fa000 + 290296
14  MPSNDArray                    	       0x199450400 0x1993fa000 + 353280
15  MPSNDArray                    	       0x19943f814 0x1993fa000 + 284692
16  MPSNDArray                    	       0x199440df8 0x1993fa000 + 290296
17  MetalPerformanceShadersGraph  	       0x1f9fd2a10 0x1f9deb000 + 1997328
18  MetalPerformanceShadersGraph  	       0x1f9f0f9e4 0x1f9deb000 + 1198564
19  MetalPerformanceShadersGraph  	       0x1f9f5e1b0 0x1f9deb000 + 1520048
20  MetalPerformanceShadersGraph  	       0x1f9f56e18 0x1f9deb000 + 1490456
21  MetalPerformanceShadersGraph  	       0x1f9f7fea0 0x1f9deb000 + 1658528
22  MetalPerformanceShadersGraph  	       0x1f9f7fce4 0x1f9deb000 + 1658084
23  MetalPerformanceShadersGraph  	       0x1f9f55aac 0x1f9deb000 + 1485484
24  MetalPerformanceShadersGraph  	       0x1fa0105e0 0x1f9deb000 + 2250208
25  MetalPerformanceShadersGraph  	       0x1fa00f9f0 0x1f9deb000 + 2247152
26  MetalPerformanceShadersGraph  	       0x1f9ec82b4 0x1f9deb000 + 905908
27  MetalPerformanceShadersGraph  	       0x1f9ec77b0 0x1f9deb000 + 903088
28  libtorch_cpu.dylib            	       0x284c2ce10 invocation function for block in at::mps::MPSStream::executeMPSGraph(MPSGraph*, NSDictionary*, NSDictionary*, at::mps::SyncType) + 152
29  libdispatch.dylib             	       0x18d003910 _dispatch_client_callout + 20
30  libdispatch.dylib             	       0x18d012cc4 _dispatch_lane_barrier_sync_invoke_and_complete + 56
31  libtorch_cpu.dylib            	       0x284c2cd60 at::mps::MPSStream::executeMPSGraph(MPSGraph*, NSDictionary*, NSDictionary*, at::mps::SyncType) + 160
32  libtorch_cpu.dylib            	       0x284cb9fe0 at::native::structured_mse_loss_out_mps::impl(at::Tensor const&, at::Tensor const&, long long, at::Tensor const&) + 1460
33  libtorch_cpu.dylib            	       0x282081dbc c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor (at::Tensor const&, at::Tensor const&, long long), &at::(anonymous namespace)::wrapper_MPS_mse_loss(at::Tensor const&, at::Tensor const&, long long)>, at::Tensor, c10::guts::typelist::typelist<at::Tensor const&, at::Tensor const&, long long>>, at::Tensor (at::Tensor const&, at::Tensor const&, long long)>::call(c10::OperatorKernel*, c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, long long) + 152
34  libtorch_cpu.dylib            	       0x28306a684 c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor (c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, long long), &torch::autograd::VariableType::(anonymous namespace)::mse_loss(c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, long long)>, at::Tensor, c10::guts::typelist::typelist<c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, long long>>, at::Tensor (c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, long long)>::call(c10::OperatorKernel*, c10::DispatchKeySet, at::Tensor const&, at::Tensor const&, long long) + 1288
35  libtorch_cpu.dylib            	       0x280f864fc at::_ops::mse_loss::call(at::Tensor const&, at::Tensor const&, long long) + 300
36  libtorch_python.dylib         	       0x126c678bc torch::autograd::THPVariable_mse_loss(_object*, _object*, _object*) + 784
37  Python                        	       0x1015793b4 cfunction_call + 60
```

### Versions

PyTorch version: 2.1.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.1.1 (arm64)
GCC version: Could not collect
Clang version: 15.0.0 (clang-1500.0.40.1)
CMake version: version 3.27.1
Libc version: N/A

Python version: 3.11.4 (v3.11.4:d2340ef257, Jun  6 2023, 19:15:51) [Clang 13.0.0 (clang-1300.0.29.30)] (64-bit runtime)
Python platform: macOS-14.1.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] numpy==1.26.2
[pip3] torch==2.1.1
[conda] Could not collect

**Comment by przem8k** [comment]
Uhm, I think I was just holding this wrong, after changing the type of the labels type to float, ie. changing the data to:

```
sample_data = """input,labels
bazinga,0.0
please-just-work,1.0
"""
```

this now works! Closing, apologies for the confusion

**Comment by przem8k** [comment]
Filed a transformers issue instead suggesting better validation higher up the stack: https://github.com/huggingface/transformers/issues/27707
